### PR TITLE
Fix `staticcheck` warnings | Resolve all but one `TODO` comments

### PR DESCRIPTION
**Pull request by taimoorgit**
## Issues

Part of: https://github.com/OpsLevel/team-platform/issues/186

## Changelog

- [x] Fix U1000's - unused stuff
- [x] Fix ST1005 - don't capitalize error messages
- [x] Change `if cond == true` to become `if cond`
- [x] Service `lifecycle_alias` and `tier_alias` should actually be mandatory
- [x] The `existingTags` var in `reconcileTags()` is never used, staticcheck complained about this.
- [x] Remove `TODO` in team - `managed_aliases` bug does not affect Team.
- [x] Remove another `TODO` in Service because the provider doesn't support `ValidateFunc` on a type list.
- [x] Make a `changie` entry

## Tophatting

I tophatted the `reconcileTags()` change to make sure `lifecycle_alias` and `tier_alias` were actually required and to make sure that `existingTags` wasn't a used variable. It worked.

```
resource "opslevel_service" "redbull" {
  name = "__ testing the tags in tf"
  lifecycle_alias = "pre-alpha"
  tier_alias = "tier_4"
  # step 1: create with no tags OR create with first tags 0,1 and skip to step 3
  # step 2: add tags 0,1 and terraform apply
  # step 3: remove tag 0, add tag 2 and terraform apply
  # all steps worked - diffs generated by terraform looked good
  tags = ["hello:world", "was:here", "removed:first"]
}
```

For `managed_aliases` tophatting on Team resource:

```
resource "opslevel_team" "redbull_racing" {
  name = "redbull racing"
  responsibilities = "idk"
  # step 1: use only this alias - terraform apply
  # step 2: run terraform apply again
  # terraform does not complain and try to add in the slug (redbull_racing)
  # i can still get the team in CLI using 'rb' or slug 'redbull_racing'
  # TL;DR managed_aliases is not necessary on the team resource despite
  # the API providing it because it doesn't suffer from the slug glitch.
  aliases = ["rb"]
  parent = "platform"
}
```

### CODE REVIEW

File: opslevel/resource_opslevel_service.go [hunk 1]
```diff
@@ -101,26 +100,11 @@ func resourceService() *schema.Resource {
 				ForceNew:    false,
 				Optional:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
-				// ValidateFunc: validateServiceTags, // TODO: Not Yet Supported
```
Comment (taimoorgit) [inline opslevel/resource_opslevel_service.go:104]: This `TODO` item isn't valid because `ValidateFunc` does not work on `schema.TypeList`. Source:

```
// ValidateFunc is honored only when the schema's Type is set to TypeInt,
// TypeFloat, TypeString, TypeBool, or TypeMap. It is ignored for all other types.
ValidateFunc [SchemaValidateFunc](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema@v1.17.2#SchemaValidateFunc)
```

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/schema@v1.17.2#Schema.ValidateFunc

Comment (taimoorgit) [reply opslevel/resource_opslevel_service.go:104]: There already exists examples of adding tags to services that the user can refer to. If the list is invalid they will get an API error anyways.

[Examples](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/resources/service)
